### ISSUE TITLE
two camera recording of raw data

### ISSUE THREAD

**Issue by duperon** [Innov-Plus/innov-ueye#2]
When recording with one camera for 3 min at 60 FPS, the folder contains 10809 files (as expected), totalling 45.3 Go

When rcording with two cameras for 3 min at60 FPS, it sometimes takes quite a long time to finish saving the files after the recording is finished.
The folder for the camera selected in the UI contains 10800 files (as expected) totalling 45 Go
The folder for the camera not selected in the UI contains 5400 files totalling 22.6 Go

With losless compression, it froze and I was not able to stop the recording.
When I looked at the files after 1min, the selected camera file was 12Go and the non-selected camera file was 5Go

**Comment by Innov-Plus** [comment]
Hi,

Are two cameras both run at 60 FPS ?

**Comment by duperon** [comment]
Yes, both are configure with a pixel clock of 256 and running a 60 fps

**Comment by Innov-Plus** [comment]
I don't have any camera with 60 FPS so I can not reproduce your problem. I'm sorry.

I test with 2 cameras I have (one 30 FPS and one 20 FPS). For now I don't have any problem of software freezing.

I saw that you forked my repo and and your forked repo is at the initial commit so I wonder if you get the latest version ? Do you pull directly from my repo ?

**Comment by duperon** [comment]
yes I didn't use the forked version, directly your version.
And you have at least one 60fps camera since you have the same one I have! (UI-3370-CP)
to get FPS up to 60 you need to change the pixel clock to 256 first (it limits the FPS)
I was wondering from what you said yesterday: You have data acquisition for each camera on a separate thread but are each saving functions on the same tread as the associated camera or are all the saving functions on the same thread? from the results I get I have the impression all the frame saving is on the same thread as the UI but I might be wrong.

**Comment by Innov-Plus** [comment]
Thank for your information. I will test it with 60fps.

In general, each camera receives its frames in its own thread and it saves the frame in the same thread that it uses to get the frame.

**Comment by Innov-Plus** [comment]
I improved the performance of the software. Please recheck if it can handle multiple cameras with 60FPS.

Thanks

**Comment by duperon** [comment]
Hi,
I can't try right now since the computer has been shipped but I will try as soon as possible.
